bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/AbstractBigtableTable.java

```diff
@@ -276,15 +277,39 @@ public Result get(Get get) throws IOException {
   @Override
   public ResultScanner getScanner(final Scan scan) throws IOException {
     LOG.trace("getScanner(Scan)");
-    Span span = TRACER.spanBuilder("BigtableTable.scan").startSpan();
+    final Span span = TRACER.spanBuilder("BigtableTable.scan").startSpan();
     try (Scope scope = TRACER.withSpan(span)) {
 
       final ResultScanner scanner = clientWrapper.readRows(hbaseAdapter.adapt(scan));
       if (hasWhileMatchFilter(scan.getFilter())) {
         return Adapters.BIGTABLE_WHILE_MATCH_RESULT_RESULT_SCAN_ADAPTER.adapt(scanner, span);
       }
-      // TODO: need to end the span when stream ends
-      return scanner;
+
+      return new AbstractClientScanner() {
+        @Override
+        public Result next() throws IOException {
+          Result row = scanner.next();
+          if (row == null) {
+            // Null signals EOF.
+            span.end();
+            return null;
+          }
+          return row;
+        }
+
+        @Override
+        public void close() {
+          try {
+            scanner.close();
+          } finally {
+            span.end();
+          }
+        }
+
+        public boolean renewLease() {
```

Comment:
Is This method is really used?

Reply:
This method is unlikely used by anyone, but it must be present because it's an interface method.